unlock-app/src/components/interface/locks/Settings/forms/UpdateHooksForm.tsx

```diff
@@ -36,14 +39,70 @@ interface HookValueProps {
   hookName: HookName
 }
 
+const isValidAddress = (address?: string) => {
+  return address?.length ? ethers.utils.isAddress(address) : true
+}
+
+interface CustomComponentProps {
+  name: string
+  disabled: boolean
+  selectedOption?: string
+}
+
+const CustomContractHook = ({
```

Comment:
I thinl we could move this to its own file in a `hooks/` folder but you can do that in a later PR.